viewer/packages/rendering/src/ResizeHandler.ts

```diff
@@ -0,0 +1,143 @@
+/*!
+ * Copyright 2022 Cognite AS
+ */
+
+import { WebGLRenderer, PerspectiveCamera, Vector2 } from 'three';
+
+import { CameraManager } from '@reveal/camera-manager';
+import assert from 'assert';
+
+export type ResizeHandlerOptions = {
+  renderResolutionThreshold?: number;
+};
+
+export class ResizeHandler {
+  private readonly _defaultResolutionThreshold = 1.4e6;
+
+  private readonly _renderer: WebGLRenderer;
+  private readonly _cameraManager: CameraManager;
+
+  private _stoppedCameraResolutionThreshold: number;
+  private _currentResolutionThreshold: number;
+  private _movingResolutionFactor: number = 1;
+
+  private readonly _onCameraChangeCallback: () => void;
+  private readonly _onCameraStopCallback: () => void;
+
+  private readonly _resizeObserver: ResizeObserver | undefined;
+
+  private _shouldResize: boolean = false;
+
+  constructor(renderer: WebGLRenderer, cameraManager: CameraManager, resizeOptions?: ResizeHandlerOptions) {
+    this._stoppedCameraResolutionThreshold =
+      resizeOptions?.renderResolutionThreshold ?? this._defaultResolutionThreshold;
+    this._currentResolutionThreshold = this._stoppedCameraResolutionThreshold;
+
+    this._resizeObserver = this.setupResizeListener(renderer);
+    this._cameraManager = cameraManager;
+
+    this._onCameraChangeCallback = () => {
+      this._currentResolutionThreshold = this._movingResolutionFactor * this._stoppedCameraResolutionThreshold;
+      this._shouldResize = true;
+    };
+    this._onCameraStopCallback = () => {
+      this._currentResolutionThreshold = this._stoppedCameraResolutionThreshold;
+      this._shouldResize = true;
+    };
+
+    this._renderer = renderer;
+  }
+
+  public get needsRedraw(): boolean {
+    return this._shouldResize;
+  }
+
+  public resetRedraw(): void {
+    this._shouldResize = false;
+  }
+
+  public setResolutionThreshold(threshold: number): void {
+    this._stoppedCameraResolutionThreshold = threshold;
+    this._currentResolutionThreshold = threshold;
+    this._shouldResize = true;
+  }
+
+  public setMovingCameraResolutionFactor(factor: number): void {
+    this._shouldResize = true;
+
+    if (factor === this._movingResolutionFactor) {
+      return;
+    }
+
+    assert(factor > 0 && factor <= 1);
```

Comment:
Replace with a proper `throw Error(<msg>)` - assert is only for stuff you think cannot happen and not to user errors. See https://stackoverflow.com/a/13832636. 

Also, add unit test for this.